GoMeet/src/main/webapp/test/spec/spec-email-dao.js

```diff
@@ -0,0 +1,99 @@
+describe ('EmailDAO - inviteGuestsToMeeting', function() {
+  const SENT_RESULT = ['sent', true]; 
+  const ERROR_RESPONSE = {
+    status: 404, 
+    messsage: 'Some error message'
+  };
+  const RESPONSE_INIT = {method: 'POST'};
+  const MEETING_EVENT_ID = 'qwerty12345'; 
+  // Note: Guest list should be of type object and each element within the list
+  // should be of type string 
+  const GUEST_LIST = ['guest1@gmail.com', 'guest2@gmail.com', 'guest3@gmail.com'];
+  const BAD_GUEST_LIST_1 = 'guest1@gmail.com, guest2@gmail.com, guest3@gmail.com'; 
+  const BAD_GUEST_LIST_2 = [123, 456, 789]; 
+  const BAD_GUEST_LIST_3 = ['guest1@gmail.com', 123, 'guest3@gmail.com']; 
+  const QUERY_STRING = '?meetingEventId=' + encodeURIComponent(MEETING_EVENT_ID) + 
+      '&guestList=' + encodeURIComponent(GUEST_LIST); 
+  
+  beforeEach(function() {
+    spyOn(window, 'fetch').and.callFake(async function(url) {
+      // Assume all fetches other than the hardcoded const QUERY_STRING 
+      // to be invalid
+      if (url !== EmailDAO.endpoint + QUERY_STRING) {
+        return new Response(JSON.stringify(ERROR_RESPONSE), null);
+      } else {
+        return new Response(JSON.stringify(SENT_RESULT), null);
+      }
+    }); 
+  }); 
+  
+  it('Throws an exception if the meetingEventId param is not provided', async function () {
+    let errorMessage;
+    try {
```

Comment:
Optional: you could extract the test for the error responses to a method, so that the code is less repeated, like what I did in `[spec-vote-meeting-time-dao.js](https://github.com/googleinterns/step271-2020/blob/main/GoMeet/src/main/webapp/test/spec/spec-vote-meeting-time-dao.js)`

Reply:
Done thanks for the suggestion!